packages/host/app/services/operator-mode-state-service.ts

```diff
@@ -61,20 +73,42 @@ export default class OperatorModeStateService extends Service {
     this.schedulePersist();
   }
 
+  popItemFromStack(stackIndex: number) {
+    let stack = this.state.stacks[stackIndex];
+    if (!stack) {
+      throw new Error(`No stack at index ${stackIndex}`);
+    }
+    let item = stack.pop();
+    if (!item) {
+      throw new Error(`No items in stack at index ${stackIndex}`);
+    }
+    this.schedulePersist();
+    return item;
+  }
+
   replaceItemInStack(item: StackItem, newItem: StackItem) {
     let stackIndex = item.stackIndex;
-    let itemIndex = this.state.stacks[stackIndex].items.indexOf(item);
+    let itemIndex = this.state.stacks[stackIndex].indexOf(item);
 
     if (newItem.stackIndex !== stackIndex) {
-      this.removeItemFromStack(item);
-      this.addItemToStack(newItem);
-      return this.schedulePersist();
+      // this could be a smell that the stack index should not live in the item
+      throw new Error(
+        'cannot move stack item to different stack--this can destabilize contained card pointers'
+      );
     }
 
-    this.state.stacks[stackIndex].items.splice(itemIndex, 1, newItem);
+    this.state.stacks[stackIndex].splice(itemIndex, 1, newItem);
     this.schedulePersist();
   }
 
+  shiftStack(stackItems: StackItem[], destinationIndex: number) {
+    stackItems.forEach((item) => {
+      this.removeItemFromStack(item);
```

Comment:
If we called `removeItemFromStack` as `popItemsOffStack`. It's clear that this logic doesn't make sense.

Reply:
This function also seems very "open". If you rely on user to pass list of stackItems. Are we assuming that all items come from the same stack?? I'd prefer pointing to a stack by stack index not by items.

Reply:
I had an implementation for this [here](https://github.com/cardstack/boxel/blob/77418bdd6cb30f9b1b22f02e02a73c5018db5379/packages/host/app/services/operator-mode-state-service.ts#L102C1-L112C4). I can merge my suggested changes if preferred

Reply:
I’m ok with your update except that the remove item that you have should not be allowed to remove a stack item if there is a contains pointing at it. (And keep in mind that a contains can point at another contains).

Reply:
(Probably that is a legit critique for this PR too, as well as, the remove item needs to adjust contains pointers if there are contained items effected by the remove)

Reply:
If we say that popping is the only permitted way to remove a stack item, I think that affects your suggested update as well—there would not be a remove condition in thr move stack since that could destabilize contains pointers.

Reply:
@tintinthong also I think it would just be simpler for you to push your change into this branch. (Or I can just do it—it’d only take a minute).

Reply:
> I’m ok with your update except that the remove item that you have should not be allowed to remove a stack item if there is a contains pointing at it. (And keep in mind that a contains can point at another contains).

Agreed. My implementation has that risk when considering the function `removeItemsIf`.

Reply:
Lets get this merged in and I will rebase my PR on main

Reply:
We can handle the usage of popping in another PR